### PR TITLE
Rewrite payload constraint validation BBEs

### PR DESCRIPTION
**Pull request by TharmiganK**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: examples/http-client-payload-constraint-validation/http_client_payload_constraint_validation.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # HTTP client - Payload constraint validation
 
-Through client payload constraint validation, the response payload can be validated according to the defined constraints. The constraint validation happens along with the data binding step. If the validation fails, a `http:PayloadValidationError` will be returned with the validation details.
+The client level response data binding also validates the payload against any constraints defined on the response payload type. If the validation fails, `http:PayloadValidationError` will be returned with the validation error message. Use this to directly validate the response payload with respect to the defined constraints.
```
Comment (TharmiganK) [inline examples/http-client-payload-constraint-validation/http_client_payload_constraint_validation.md:3]:
```suggestion
The client-level response data binding also validates the payload against any constraints defined on the response payload type. If the validation fails, `http:PayloadValidationError` will be returned with the validation error message. Use this to directly validate the response payload with respect to the defined constraints.
```

File: examples/http-service-payload-constraint-validation/http_service_payload_constraint_validation.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
 # REST service - Payload constraint validation
 
-Through service payload constraint validation, the request payload can be validated according to the defined constraints. The constraint validation happens along with the data binding step in the resource signature parameter. If the validation fails, a `400 Bad Request` response will be returned to the client with the validation details.
+The request payload data binding also validates the payload against any constraints defined on the request payload type. If the validation fails, a 400 Bad Request response will be returned to the client with the validation error message and the resource method execution will be skipped. Use this to directly validate the request payload with respect to the defined constraints.
```
Comment (TharmiganK) [inline examples/http-service-payload-constraint-validation/http_service_payload_constraint_validation.md:3]:
```suggestion
The request payload data binding also validates the payload against any constraints defined on the request payload type. If the validation fails, a 400 Bad Request response will be returned to the client with the validation error message. Use this to directly validate the request payload with respect to the defined constraints.
```

**Review by praneesha — APPROVED** [review]

File: examples/http-client-payload-constraint-validation/http_client_payload_constraint_validation.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # HTTP client - Payload constraint validation
 
-Through client payload constraint validation, the response payload can be validated according to the defined constraints. The constraint validation happens along with the data binding step. If the validation fails, a `http:PayloadValidationError` will be returned with the validation details.
+The client-level response data binding also validates the payload against any constraints defined on the response payload type. If the validation fails, `http:PayloadValidationError` will be returned with the validation error message. Use this to directly validate the response payload with respect to the defined constraints.
```
Comment (shafreenAnfar) [inline examples/http-client-payload-constraint-validation/http_client_payload_constraint_validation.md:3]: How about a description like the below?

The Ballerina constraint package allows you to add additional constraints to the response payload. `http:Client` uses the constraint package to validate the payload against the given constraints. Validation happens soon after the successful data-binding of the payload. Constraints can be added to a given data-type using different annotations. If the validation fails, `http:PayloadValidationError` will be returned with the validation error message. Use this to validate the response payload as the application program receives it, which allows you to guard against unnecessary processing and malicious payloads.

Comment (TharmiganK) [reply examples/http-client-payload-constraint-validation/http_client_payload_constraint_validation.md]: Changed the description of both BBEs in [a5ba91f](https://github.com/ballerina-platform/ballerina-distribution/pull/3743/commits/a5ba91fab573537e4d7a0cd632d9b7722f3c19bf)
Please review

File: examples/http-service-payload-constraint-validation/http_service_payload_constraint_validation.md [hunk 4]
```diff
@@ -1,18 +1,20 @@
 # REST service - Payload constraint validation
 
-Through service payload constraint validation, the request payload can be validated according to the defined constraints. The constraint validation happens along with the data binding step in the resource signature parameter. If the validation fails, a `400 Bad Request` response will be returned to the client with the validation details.
+The Ballerina `constraint` package allows you to add additional constraints to the response payload. The `http` resource method uses the `constraint` package to validate the payload against the given constraints. This validation happens soon after the successful data-binding of the request payload before executing the resource method. 
+The constraints can be added to a given data type using different annotations. If the validation fails, a 400 Bad Request response will be returned to the client with the validation error message.
```
Comment (TharmiganK) [inline examples/http-service-payload-constraint-validation/http_service_payload_constraint_validation.md:4]:
```suggestion
The constraints can be added to a given data type using different annotations. If the validation fails, a `400 Bad Request` response will be returned to the client with the validation error message. 
```

**Review by praneesha — APPROVED** [review]

File: examples/http-client-payload-constraint-validation/http_client_payload_constraint_validation.md [hunk 5]
```diff
@@ -1,6 +1,8 @@
 # HTTP client - Payload constraint validation
 
-Through client payload constraint validation, the response payload can be validated according to the defined constraints. The constraint validation happens along with the data binding step. If the validation fails, a `http:PayloadValidationError` will be returned with the validation details.
+The Ballerina `constraint` package allows you to add additional constraints to the response payload. The `http:Client` uses the `constraint` package to validate the payload against the given constraints. This validation happens soon after the successful data-binding of the response payload.
```
Comment (TharmiganK) [inline examples/http-client-payload-constraint-validation/http_client_payload_constraint_validation.md:3]:
```suggestion
The Ballerina `constraint` module allows you to add additional constraints to the response payload. The `http:Client` uses the `constraint` module to validate the payload against the given constraints. This validation happens soon after the successful data-binding of the response payload. 
```

File: examples/http-client-payload-constraint-validation/http_client_payload_constraint_validation.md [hunk 6]
```diff
@@ -14,4 +16,4 @@ Run the client program by executing the following command.
 ## Related links
 - [`http` package - API documentation](https://lib.ballerina.io/ballerina/http/latest/)
```
Comment (TharmiganK) [inline examples/http-client-payload-constraint-validation/http_client_payload_constraint_validation.md:17]:
```suggestion
- [`http` module - API documentation](https://lib.ballerina.io/ballerina/http/latest/)
```

File: examples/http-client-payload-constraint-validation/http_client_payload_constraint_validation.md [hunk 7]
```diff
@@ -14,4 +16,4 @@ Run the client program by executing the following command.
...
 - [`http` package - Specification](/spec/http/)
```
Comment (TharmiganK) [inline examples/http-client-payload-constraint-validation/http_client_payload_constraint_validation.md:18]:
```suggestion
- [`http` module - Specification](/spec/http/)
```

File: examples/http-service-payload-constraint-validation/http_service_payload_constraint_validation.md [hunk 8]
```diff
@@ -1,18 +1,20 @@
 # REST service - Payload constraint validation
 
-Through service payload constraint validation, the request payload can be validated according to the defined constraints. The constraint validation happens along with the data binding step in the resource signature parameter. If the validation fails, a `400 Bad Request` response will be returned to the client with the validation details.
+The Ballerina `constraint` package allows you to add additional constraints to the response payload. The `http` resource method uses the `constraint` package to validate the payload against the given constraints. This validation happens soon after the successful data-binding of the request payload before executing the resource method.
```
Comment (TharmiganK) [inline examples/http-service-payload-constraint-validation/http_service_payload_constraint_validation.md:3]:
```suggestion
The Ballerina `constraint` module allows you to add additional constraints to the response payload. The `http` resource method uses the `constraint` module to validate the payload against the given constraints. This validation happens soon after the successful data-binding of the request payload before executing the resource method. 
```

File: examples/http-service-payload-constraint-validation/http_service_payload_constraint_validation.md [hunk 9]
```diff
@@ -1,18 +1,20 @@
...
 
 ::: out http_service_payload_constraint_validation.server.out :::
 
-Invoke the service by executing the following cURL command in a new terminal. Here, an album which exceeds the constraints are sent to the service.
+Invoke the service by executing the following cURL command in a new terminal. Here, an album with a lengthy title is sent to the service.
 
 ::: out http_service_payload_constraint_validation.client.out :::
 
 ## Related links
 - [`http` package - API documentation](https://lib.ballerina.io/ballerina/http/latest/)
```
Comment (TharmiganK) [inline examples/http-service-payload-constraint-validation/http_service_payload_constraint_validation.md:18]:
```suggestion
- [`http` module - API documentation](https://lib.ballerina.io/ballerina/http/latest/)
```

File: examples/http-service-payload-constraint-validation/http_service_payload_constraint_validation.md [hunk 10]
```diff
@@ -1,18 +1,20 @@
...
 - [`http` package - Specification](/spec/http/)
```
Comment (TharmiganK) [inline examples/http-service-payload-constraint-validation/http_service_payload_constraint_validation.md:19]:
```suggestion
- [`http` module - Specification](/spec/http/)
```

**Pull request merged**